sentinel-dashboard/src/main/java/com/alibaba/csp/sentinel/dashboard/controller/v2/ParamFlowRuleControllerV2.java

```diff
@@ -0,0 +1,271 @@
+import com.alibaba.csp.sentinel.dashboard.domain.Result;
+import com.alibaba.csp.sentinel.dashboard.repository.rule.RuleRepository;
+import com.alibaba.csp.sentinel.dashboard.rule.DynamicRuleProvider;
+import com.alibaba.csp.sentinel.dashboard.rule.DynamicRulePublisher;
+import com.alibaba.csp.sentinel.dashboard.util.VersionUtils;
+import com.alibaba.csp.sentinel.slots.block.RuleConstant;
+import com.alibaba.csp.sentinel.util.StringUtil;
+import java.util.Date;
+import java.util.List;
+import java.util.Optional;
+import java.util.concurrent.ExecutionException;
+import javax.servlet.http.HttpServletRequest;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+/**
+ * @author threedr3am
+ */
+@RestController
+@RequestMapping(value = "/v2/paramFlow")
+public class ParamFlowRuleControllerV2 {
+
+    private final Logger logger = LoggerFactory.getLogger(ParamFlowRuleControllerV2.class);
+
+    @Autowired
+    private AppManagement appManagement;
+    @Autowired
+    private RuleRepository<ParamFlowRuleEntity, Long> repository;
+
+    @Autowired
+    private AuthService<HttpServletRequest> authService;
+
+    @Autowired
+    @Qualifier("paramFlowRuleDefaultProvider")
+    private DynamicRuleProvider<List<ParamFlowRuleEntity>> ruleProvider;
+    @Autowired
+    @Qualifier("paramFlowRuleDefaultPublisher")
+    private DynamicRulePublisher<List<ParamFlowRuleEntity>> rulePublisher;
+
+    private boolean checkIfSupported(String app, String ip, int port) {
+        try {
+            return Optional.ofNullable(appManagement.getDetailApp(app))
+                .flatMap(e -> e.getMachine(ip, port))
+                .flatMap(m -> VersionUtils.parseVersion(m.getVersion())
+                    .map(v -> v.greaterOrEqual(version020)))
+                .orElse(true);
+            // If error occurred or cannot retrieve machine info, return true.
+        } catch (Exception ex) {
+            return true;
+        }
+    }
+
+    @GetMapping("/rules")
+    public Result<List<ParamFlowRuleEntity>> apiQueryAllRulesForMachine(HttpServletRequest request,
+                                                                        @RequestParam String app,
+                                                                        @RequestParam String ip,
```

Comment:
Here the `ip` and `port` parameter are not needed here. And we don't have to check versions here.